schema-registry/common/src/main/java/com/hortonworks/registries/schemaregistry/retry/RetryContext.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2016-2019 Cloudera, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hortonworks.registries.schemaregistry.retry;
+
+import com.hortonworks.registries.schemaregistry.retry.policy.NOOPRetryPolicy;
+import com.hortonworks.registries.schemaregistry.retry.policy.RetryPolicy;
+import com.hortonworks.registries.schemaregistry.retry.request.Request;
+
+import java.util.Objects;
+
+public class RetryContext<T> {
+
+    private RetryPolicy policy = new NOOPRetryPolicy();
+
+    private Request<T> request;
+
+    private Class<? extends RuntimeException> exceptionClass = RuntimeException.class;
+
+    public static class Builder<T> {
+
+        private RetryContext<T> retryContext = new RetryContext<T>();
```

Comment:
One can reuse the same `Builder` instance in multiple places. Better to create the `RetryContext` inside the `build` method.

Reply:
Yep, makes sense.